pkg/network/vmicontroller/status.go

```diff
@@ -34,15 +34,49 @@ import (
 )
 
 func UpdateStatus(vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) error {
-	interfaceStatuses, err := calculateSecondaryIfaceStatuses(vmi, multus.NetworkStatusesFromPod(pod))
+	var interfaceStatuses []v1.VirtualMachineInstanceNetworkInterface
+
+	networkStatuses := multus.NetworkStatusesFromPod(pod)
+
```

Comment:
nit: consider changing the computation order of `calculatePrimaryIfaceStatus` and `calculateSecondaryIfaceStatus` as the worst case scenario of the latter yields an error and voids the result of the former.

Reply:
There is an importance to the ordering here:
We want the primary interface to be first in the interface statuses slice, since its IP value is considered to be the VMI's IP address:
https://github.com/kubevirt/kubevirt/blob/ca0227a4c9aee3e6ad16b449ee57bea5325ba998/pkg/virt-operator/resource/generate/components/crds.go#L170

There is similar logic in the virt-handler's network status reporting logic.

Reply:
Our ability to read and compute the flow of changes is also more important than the optimization of a few cycles that may be saved if an error will occur.
I would advice never to prefer that over readability.

BTW, without entering a long debate here, both the compiler or CPU may process this in a different order, use cashes and many other techniques it may find fit.
The code better express the intent and less be worried about optimizations like these.
Said that, I am many times bothered if a search is done N times instead of once, so there is never a total truth to this game.

Reply:
Thanks for the clarification, I concur that premature optimization should be avoided in general, especially in favor of readability.